internal/node/component/decentralized/component.go

```diff
@@ -28,13 +29,21 @@ type Component struct {
 
 const Name = "decentralized"
 
+const MaxRecentRequests = 10
+
+var (
+	RecentRequests *cb.Queue
```

Comment:
The queue is not thread-safe, so you also need to add a `RWMutex` to it here.

Reply:
We should avoid accessing the variable directly, and instead encapsulate read and write functions for it to be used in other packages to avoid forgetting to use its `RWMutex`.